pkg/util/strings/strings.go

```diff
@@ -33,62 +34,17 @@ func Substring(s string, i int, length int) string {
 	return s[i : i+length]
 }
 
-// IndexInSlice returns the index of a string element in a given slice
-func IndexInSlice(arr []string, val string) int {
-	for i, v := range arr {
-		if v == val {
-			return i
-		}
-	}
-	return -1
-}
-
-// CloneBoolMap returns an exact copy of a map consisting string key and bool value
-func CloneBoolMap(in map[string]bool) map[string]bool {
-	if in == nil {
-		return nil
-	}
-	out := make(map[string]bool)
-	for k, v := range in {
-		out[k] = v
-	}
-	return out
-}
-
 // Equal returns true if the slices contain identical values in the identical order
 func Equal(s1, s2 []string) bool {
-	if s1 == nil && s2 == nil {
-		return true
-	}
-	if s1 == nil || s2 == nil || len(s1) != len(s2) {
-		return false
-	}
-	for i, v := range s1 {
-		if v != s2[i] {
-			return false
-		}
-	}
-	return true
+	return slices.Equal(s1, s2)
```

Comment:
we should just remove this func and its unit tests altogether, and change the one reference to it (in proxy/tls/options/options.go) over to slices.Equal